src/components/NoteEditor/NoteEditor.js

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { useState } from "react";
+import { Bookmark, Color, Add } from "../../assets";
+import { Description } from "./Editor/Description";
+import { Title } from "./Editor/Title";
+import { AddLabel } from "./Label/AddLabel";
+import { Lables } from "./Label/Lables";
+import "./noteEditor.css";
+import { Pallete } from "./Pallete/Pallete";
+import { Priority } from "./Priority/Priority";
+
+export const NoteEditor = ({ addNote, noteState, noteDispatch }) => {
+  const [pallete, setPallete] = useState(false);
+  const [lable, setLable] = useState(false);
+  const { color, title, description, lables } = noteState;
+
+  const toggle = (setState, toggleState) => {
+    toggleState(false);
+    setState((prev) => !prev);
+  };
+
+  const addToNotes = () => {
+    if (title.trim(" ").length > 0 || description.trim(" ").length > 0) {
+      addNote();
+    } else {
+      alert("Enter Note");
+    }
+    setLable(false);
+    setPallete(false);
+  };
+
+  return (
+    <div style={{ background: color }} className='note-editor card-container'>
+      <Title title={title} noteDispatch={noteDispatch} />
+      <Description description={description} noteDispatch={noteDispatch} />
+      <Lables lables={lables} noteDispatch={noteDispatch} />
+      <div className='note-action-container flex jc-between ai-center'>
+        <div className='note-btn flex flex-gap'>
+          <div
+            onClick={(e) => toggle(setPallete, setLable)}
+            className='btn btn-icon'>
+            <img src={Color} alt='color-pallete' />
+          </div>
+          <div
+            onClick={(e) => toggle(setLable, setPallete)}
+            className='btn btn-icon'>
+            <img src={Bookmark} alt='color-pallete' />
+          </div>
+          <Priority noteDispatch={noteDispatch} />
+        </div>
+        <div onClick={addToNotes} className='note-add-btn  btn btn-icon'>
+          <img src={Add} alt='add-note' />
+        </div>
+      </div>
+      {pallete && <Pallete noteDispatch={noteDispatch} />}
```

Comment:
could have used ternary operator instead

Reply:
I only want my element to render when the conditon true otherwise render nothing